packages/twenty-server/src/workspace/calendar/repositories/calendar-event/calendar-event.service.ts

```diff
@@ -0,0 +1,218 @@
+      `SELECT id, "iCalUID" FROM ${dataSourceSchema}."calendarEvent" WHERE "iCalUID" = ANY($1)`,
+      [iCalUIDs],
+      workspaceId,
+      transactionManager,
+    );
+
+    const iCalUIDsCalendarEvnetIdsMap = new Map<string, string>();
+
+    calendarEvents.forEach((calendarEvent) => {
+      iCalUIDsCalendarEvnetIdsMap.set(calendarEvent.iCalUID, calendarEvent.id);
+    });
+
+    return iCalUIDsCalendarEvnetIdsMap;
+  }
+
+  public async saveCalendarEvents(
+    calendarEvents: CalendarEvent[],
+    workspaceId: string,
+    transactionManager?: EntityManager,
+  ): Promise<void> {
+    if (calendarEvents.length === 0) {
+      return;
+    }
+
+    const dataSourceSchema =
+      this.workspaceDataSourceService.getSchemaName(workspaceId);
+
+    const valuesString = valuesStringForBatchRawQuery(calendarEvents, 14);
+
+    const values = calendarEvents.flatMap((calendarEvent) => [
+      calendarEvent.id,
+      calendarEvent.title,
+      calendarEvent.isCanceled,
+      calendarEvent.isFullDay,
+      calendarEvent.startsAt,
+      calendarEvent.endsAt,
+      calendarEvent.externalCreatedAt,
+      calendarEvent.externalUpdatedAt,
+      calendarEvent.description,
+      calendarEvent.location,
+      calendarEvent.iCalUID,
+      calendarEvent.conferenceSolution,
+      calendarEvent.conferenceUri,
+      calendarEvent.recurringEventExternalId,
+    ]);
+
+    await this.workspaceDataSourceService.executeRawQuery(
+      `INSERT INTO ${dataSourceSchema}."calendarEvent" ("id", "title", "isCanceled", "isFullDay", "startsAt", "endsAt", "externalCreatedAt", "externalUpdatedAt", "description", "location", "iCalUID", "conferenceSolution", "conferenceUri", "recurringEventExternalId") VALUES ${valuesString}`,
+      values,
+      workspaceId,
+      transactionManager,
+    );
+  }
+
+  public async updateCalendarEvents(
+    calendarEvents: CalendarEvent[],
+    workspaceId: string,
+    transactionManager?: EntityManager,
+  ): Promise<void> {
+    if (calendarEvents.length === 0) {
+      return;
+    }
+
+    const dataSourceSchema =
+      this.workspaceDataSourceService.getSchemaName(workspaceId);
+
+    const valuesString = valuesStringForBatchRawQuery(calendarEvents, 13, [
+      'text',
+      'boolean',
+      'boolean',
+      'timestamptz',
+      'timestamptz',
+      'timestamptz',
+      'timestamptz',
+      'text',
+      'text',
+      'text',
+      'text',
+      'text',
+      'text',
+    ]);
+
+    const values = calendarEvents.flatMap((calendarEvent) => [
+      calendarEvent.title,
+      calendarEvent.isCanceled,
+      calendarEvent.isFullDay,
+      calendarEvent.startsAt,
+      calendarEvent.endsAt,
+      calendarEvent.externalCreatedAt,
+      calendarEvent.externalUpdatedAt,
+      calendarEvent.description,
+      calendarEvent.location,
+      calendarEvent.iCalUID,
+      calendarEvent.conferenceSolution,
+      calendarEvent.conferenceUri,
+      calendarEvent.recurringEventExternalId,
+    ]);
```

Comment:
Same as above, feel like we can create a util that can create the value string, and create the flat map.
If we need to filter properties we want to flatten we can create a typed param base on the provided object, that allow passing an array of keys.